File: 5.Searching/ArrangingCoins.js
/**
 * @param {number} n
 * @return {number}
 */
var arrangeCoins = function(n) {
    let lo = 1;
    let hi = n;
    let ans = 1;
    while(lo <= hi) {
        let mid = lo + Math.floor((hi - lo) / 2);
        let value = Math.floor(((mid)*(mid+1))/2);
        // if(value == n) 
        if(value <= n) {
            ans = mid;
            lo = mid+1;
        } else {
            hi = mid-1;
        }
    }
    return ans;
};

File: 5.Searching/ArrangingCoins.java
public class ArrangingCoins {
        public int arrangeCoins(int n) {
            long lo = 1;
            long hi = n;
            long ans = 1;
            while (lo <= hi) {
                long mid = lo + (hi - lo) / 2;
                long value = (mid * (mid + 1)) / 2;
                if (value <= n) {
                    ans = mid;
                    lo = mid + 1;
                } else {
                    hi = mid - 1;
                }
            }
            return (int) ans;
        }
    
}
